internal/rotate.go

```diff
@@ -0,0 +1,88 @@
+package internal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+)
+
+type CertRotationState struct {
+	EstServer string
+}
+
+type CertRotationHandler struct {
+	State           CertRotationState
+	stateFile       string
+	stateFileExists bool
+	app             *App
+	client          *http.Client
+	crypto          *EciesCrypto
+	steps           []CertRotationStep
```

Comment:
Usually the  state machine refers to the current state in its context. The state in this  context is actually "step"/CertRotationStep. So, in this context, "current state" can be an index in `[]CertRotationStep`.

Reply:
i've force pushed a change to the PR that tracks a `StepIdx` which allows me to remove the `.Completed` method.